internal/services/cognitive/cognitive_account_rai_policy_resource.go

```diff
@@ -0,0 +1,396 @@
+						Required:     true,
+						ValidateFunc: validation.StringInSlice(raipolicies.PossibleValuesForRaiPolicyContentSource(), false),
+					},
+				},
+			},
+		},
+
+		"custom_blocklist": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"block_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Required: true,
+					},
+					"source": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringInSlice(raipolicies.PossibleValuesForRaiPolicyContentSource(), false),
+					},
+				},
+			},
+		},
+
+		"mode": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice(raipolicies.PossibleValuesForRaiPolicyMode(), false),
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r CognitiveAccountRaiPolicyResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"type": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (r CognitiveAccountRaiPolicyResource) ModelObject() interface{} {
+	return &AccountRaiPolicyResourceModel{}
+}
+
+func (r CognitiveAccountRaiPolicyResource) ResourceType() string {
+	return "azurerm_cognitive_account_rai_policy"
+}
+
+func (r CognitiveAccountRaiPolicyResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cognitive.RaiPoliciesClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model AccountRaiPolicyResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			cognitiveAccountId, err := raipolicies.ParseAccountID(model.AccountId)
+			if err != nil {
+				return err
+			}
+
+			id := raipolicies.NewRaiPolicyID(subscriptionId, cognitiveAccountId.ResourceGroupName, cognitiveAccountId.AccountName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			locks.ByID(cognitiveAccountId.ID())
+			defer locks.UnlockByID(cognitiveAccountId.ID())
+
+			raiPolicy := raipolicies.RaiPolicy{
+				Name: pointer.To(model.Name),
+				Properties: &raipolicies.RaiPolicyProperties{
+					BasePolicyName:   pointer.To(model.BasePolicyName),
+					ContentFilters:   expandRaiPolicyContentFilters(model.ContentFilter),
+					CustomBlocklists: expandCustomBlockLists(model.CustomBlockList),
+					Mode:             pointer.To(raipolicies.RaiPolicyMode(model.Mode)),
```

Comment:
These are Optional so we should only be setting these in the payload in the create if a user has set a value for them in their config.

Reply:
Will check exist before setting this value